compiler/rustc_middle/src/traits/mod.rs

```diff
@@ -921,6 +924,9 @@ impl ObjectSafetyViolation {
                         .into()
                 }
             }
+            ObjectSafetyViolation::SupertraitNonLifetimeBinder(_) => {
+                format!("predicate cannot reference non-lifetime `for<...>` bound variables").into()
```

Comment:
Should me maybe use a more user-friendly error message here? Can we expect users to know what predicates are?

Reply:
i would hope if they're using non-lifetime binders, but I guess i can de-types-ify the message by saying "where clause" :smile_cat: